src/App.tsx

```diff
@@ -1,34 +1,77 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.css'
+import './App.css';
+import Navbar from './components/Navbar';
+import Categories from './components/Categories';
+import ProductPage from './components/ProductPage';
+import Cart from './components/Cart';
+import { useState } from 'react';
+
+interface ProductCartItem {
+  id: string;
+  image: string;
+  title: string;
+  price: number;
+  category: string;
+  description: string;
+  quantity: number;
+}
 
 function App() {
-  const [count, setCount] = useState(0)
+  // gett current route
+  const currentPath = window.location.pathname;
+
+  // state variables
+  const [isCartOpen, setIsCartOpen] = useState(false);
+  const [cartProducts, setCartProducts] = useState<ProductCartItem[]>([]);
+  const [cartCount, setCartCount] = useState(0);
+
+  // cart handling
+  const toggleCart = () => {
+    console.log(isCartOpen);
+    setIsCartOpen(!isCartOpen);
+  };
```

Comment:
console.log 🤭. For single expression functions you can make them blockless as well
```suggestion
  const toggleCart = () => setIsCartOpen(!isCartOpen)
```